src/sentry/integrations/vercel/uninstall.py

```diff
@@ -0,0 +1,96 @@
+                extra={"configuration_id": configuration_id, "external_id": external_id},
+            )
+            return self.respond(status=404)
+
+        orgs = integration.organizations.all()
+
+        if len(orgs) == 1:
+            create_audit_entry(
+                request=request,
+                organization=integration.organizations.all()[0],
+                target_object=integration.id,
+                event=AuditLogEntryEvent.INTEGRATION_REMOVE,
+                # TODO(meredith): If we create vercel identities from the userId
+                # we could attempt to find the user in Sentry and pass that in for
+                # the actor instead.
+                actor_label="Vercel User",
+                data={"provider": integration.provider, "name": integration.name},
+            )
+            integration.delete()
+            return self.respond(status=202)
+
+        configuration = integration.metadata["configurations"].pop(configuration_id)
+        if configuration_id == integration.metadata["installation_id"]:
+            # if we are uninstalling a primary configuration, and there are
+            # multiple orgs connected to this integration we must update
+            # the crendentials (access_token, webhook_id etc)
+            next_config_id, next_config = integration.metadata["configurations"].items()[0]
+
+            integration.metadata["access_token"] = next_config["access_token"]
+            integration.metadata["webhook_id"] = next_config["webhook_id"]
+            integration.metadata["installation_id"] = next_config_id
+
+        try:
+            OrganizationIntegration.objects.get(
+                organization_id=configuration["organization_id"], integration_id=integration.id
+            ).delete()
+        except OrganizationIntegration.DoesNotExist:
+            logger.info(
+                "vercel.uninstall.missing-org-integration",
+                extra={
+                    "configuration_id": configuration_id,
+                    "external_id": external_id,
+                    "integration_id": integration.id,
+                    "organization_id": configuration["organization_id"],
+                },
+            )
+            return self.respond(status=404)
+
+        integration.save()
+
+        organization = Organization.objects.get(id=configuration["organization_id"])
+        create_audit_entry(
+            request=request,
+            organization=organization,
+            target_object=integration.id,
+            event=AuditLogEntryEvent.INTEGRATION_REMOVE,
+            # TODO(meredith): If we create vercel identities from the userId
+            # we could attempt to find the user in Sentry and pass that in for
+            # the actor instead.
+            actor_label="Vercel User",
+            data={"provider": integration.provider, "name": integration.name},
+        )
+
+        return self.respond(status=202)
```

Comment:
@MeredithAnya Are deletions happening in an async task? if not, this should be a `204`.

Reply:
they are not, will update